src/app/shared/services/user-session.service.ts

```diff
@@ -65,6 +66,14 @@ export class UserSessionService implements OnDestroy {
     return !session || session.user.tempUser;
   }
 
+  updateCurrentUser(changes: UpdateUserBody): Observable<void> {
+    return this.currentUserService.update(changes).pipe(
+      tap(() => {
+        this.refresh$.next();
+      }),
```

Comment:
I'm not sure this is the best way of doing it

Reply:
:warning: not fixed (yet)

Reply:
I have another idea on how to perform the refresh, what do you think ↓ ?
```ts
const update$ = this.currentUserService.update(changes);

of(null).pipe(
  skip(1),
  repeatWhen(() => update$),
).subscribe(() => this.refresh$.next());

return update$;
```

Reply:
I found another way to perform this `refresh$.next()`, what do you think ↓ ?
```ts
updateCurrentUser(changes: UpdateUserBody): Observable<void> {
  const update$ = this.currentUserService.update(changes).pipe(share());

  of(null).pipe(
    delayWhen(() => update$),
  ).subscribe(() => this.refresh$.next());

  return update$;
}
```

Reply:
I also tested what happens when the caller of `updateCurrentUser()` delays the subscription and it works fine, the http request is not sent until the observable is subscribed to

Reply:
Fixed